drivers/adc/ad7768/ad7768.c

```diff
@@ -626,21 +628,85 @@ int32_t ad7768_setup(ad7768_dev **device,
 	ad7768_dev *dev;
 	int32_t ret;
 
-	dev = (ad7768_dev *)no_os_malloc(sizeof(*dev));
+	ret = ad7768_setup_begin(&dev, init_param);
+	if (ret)
+		return ret;
+
+	ret = ad7768_setup_finish(dev, init_param);
+	if (ret)
+		goto error;
+
+	*device = dev;
+
+	return 0;
+error:
+	no_os_free(dev);
+
+	return -1;
+}
+
+/**
+ * Begin initializing the device.
+ * @param device - The device structure.
+ * @param init_param - The structure that contains the device initial
+ * 					   parameters.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad7768_setup_begin(ad7768_dev **device,
+			   ad7768_init_param init_param)
+{
+	ad7768_dev *dev;
+	int32_t ret;
+
+	dev = (ad7768_dev *)no_os_calloc(1, sizeof(*dev));
 	if (!dev) {
 		return -1;
 	}
 
 	ret = no_os_spi_init(&dev->spi_desc, &init_param.spi_init);
+	if (ret)
+		goto error;
 
 	dev->pin_spi_input_value = init_param.pin_spi_input_value;
 
 	dev->pin_spi_ctrl = dev->pin_spi_input_value ?
 			    AD7768_SPI_CTRL : AD7768_PIN_CTRL;
 
-	ret |= no_os_gpio_get(&dev->gpio_reset, &init_param.gpio_reset);
+	ret = no_os_gpio_get(&dev->gpio_reset, &init_param.gpio_reset);
+	if (ret)
+		goto error;
 	dev->gpio_reset_value = init_param.gpio_reset_value;
 
+	if (dev->gpio_reset) {
```

Comment:
if you call `no_os_gpio_get` instead of `no_os_gpio_get_optional` why do you need this check?

Reply:
Removed the check